netbox/dcim/search.py

```diff
@@ -41,6 +41,8 @@ class DeviceIndex(SearchIndex):
         ('asset_tag', 50),
         ('serial', 60),
         ('name', 100),
+        ('primary_ip4', 110),
```

Comment:
This will not work, primary_ip4 is a foreignkey to the IPAddress model and as such cannot be indexed. Furthermore changing indexes would probably require dumping the existing index in a migration.

Reply:
Works fine. When adding an IP to a device and checking the 'make primary' it seems to be indexed.
I can search that IP immediately and find the ip, device, prefix, aggregate

Reply:
I see why it works, the device is for some reason saved when the primary ip is modified in the UI (empty changelog though). In general we can't index related objects, because it's complex to update the index when related objects change.

In this case, try to modify the primary ip via. the API and change the address. You will end up with an inconsistent cache. Here I change the IPAddress object assigned as primary ip from 199.199.199.3/25 to 199.199.199.3/26. The index for the ip address was updated, but not for the device as none of the fields on the device model changed.

![image](https://user-images.githubusercontent.com/400797/221126161-ded06e09-40ff-4d92-bcb8-cede497634f4.png)

So no, it doesn't work.

Reply:
> Isn't a the reindex is done automatically during the housekeeping ?

No, a full reindex can take 10+ minutes on large instances and during the reindex the search index would be incomplete.

> A note can be added to run the reindex manually if needed.

Currently, as I said, the way we do it is to drop the index for the specific model in a migration and then we have added a reindex --lazy command which is run during ./upgrade.sh. This avoids having to refresh the entire index every upgrade.

Reply:
That's a bummer.. I really liked the device in those results. I hope this might make it in a different FR, as there is a request to add related objects to the results

I'll remove this from the PR